auth/aws/partitions/gen.sh

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env sh
+
+# Copyright (c) 2025 OpenBao a Series of LF Projects, LLC
+# SPDX-License-Identifier: MPL-2.0
+
+#################################################################
+# generate a map[string]string mapping from partition -> region #
+#################################################################
+
+# Find the aws module source dir, which contains a json with the data we want
+# We could probably also get the data from some API, but using only content of this repo makes it deterministic
+DIR=$(go list -f '{{.Dir}}' 'github.com/aws/aws-sdk-go-v2/internal/endpoints/awsrulesfn')
+
+TMP_GO=$(mktemp)
+
+# Generate some boilerplate
+echo '// Copyright (c) 2025 OpenBao a Series of LF Projects, LLC' > "$TMP_GO"
+echo '// SPDX-License-Identifier: MPL-2.0' >> "$TMP_GO"
+echo '' >> "$TMP_GO"
+echo "// Code generated by ${0}. DO NOT EDIT." >> "$TMP_GO"
+echo '' >> "$TMP_GO"
+echo 'package partitions' >> "$TMP_GO"
+echo 'var data = map[string]string{' >> "$TMP_GO"
+
+# Generate map entries
+jq -r '
+    .partitions[] |
+        "\(.id | @json): \(.outputs.implicitGlobalRegion | @json),"
+' "$DIR/partitions.json" >> "$TMP_GO"
+
+echo '}' >> "$TMP_GO"
```

Comment:
```suggestion
cat > "${TMP_GO}" << EOF
// Copyright (c) 2025 OpenBao a Series of LF Projects, LLC
// SPDX-License-Identifier: MPL-2.0

// Code generated by ${0}. DO NOT EDIT.

package partitions
var data = map[string]string$(jq -c  '[.partitions[]|{ "key": .id, "value": .outputs.implicitGlobalRegion} ] | from_entries' "$DIR/partitions.json")
EOF

```
I think using `cat` to template multi line strings into a file is a bit cleaner than multiple calls to `echo`.